eth/backend.go

```diff
@@ -143,8 +143,10 @@ func New(ctx *node.ServiceContext, config *Config) (*Ethereum, error) {
 
 	if !config.SkipBcVersionCheck {
 		bcVersion := rawdb.ReadDatabaseVersion(chainDb)
-		if bcVersion != core.BlockChainVersion && bcVersion != 0 {
-			return nil, fmt.Errorf("Blockchain DB version mismatch (%d / %d).\n", bcVersion, core.BlockChainVersion)
+		if bcVersion != nil && *bcVersion > core.BlockChainVersion {
+			return nil, fmt.Errorf("Blockchain DB version mismatch (%d / %d).\n", *bcVersion, core.BlockChainVersion)
```

Comment:
Since this is an error, the string should be lowercase. Also I'd expand the `/` form a bit, otherwise it's hard to interpret. Eg. `fmt.Errorf("database version is v%d, Geth %s only supports v%d", *bcVersion, params.VersionWithMeta(), core.BlockChainVersion)`